Add tests for XML writer

dstansby opened the pull request:
This should fix https://github.com/brainglobe/brainglobe-napari-io/issues/10 (the XML writers not working), and adds associated tests.

Note that the XML "multiple file" writer works fine for writing a single file to (as far as I can tell from local testing), so to reduce code duplication and the number of options in the plugin menu I've removed the "single file" writer.

adamltyson commented:
This has fixed saving of a single layer to xml, but the multiple layer writing (e.g. >1 points layer to a single XML file) still seems to be broken for me.

dstansby commented:
Thanks for testing. I've refactored the code a bit to reduce duplication, and it seems to be working for me locally. @adamltyson could you check again? The tests failing are unrelated, and fixed in https://github.com/brainglobe/brainglobe-napari-io/pull/15

adamltyson commented:
Maybe I mistyped on my last comment, because now I can save multiple layers, but not a single layer. I get:
```
No data written! There may be no plugins capable of writing these 1 layers to /home/adam/Desktop/test_single_layer.xml.
```

dstansby commented:
I can reproduce that. The multiple writer seems capable of saving data from a single layer though (at least from local tests), so I'd propose that we remove the single layer writer and have just one points writer that can hangle both single and multiple layers?

adamltyson commented on brainglobe_napari_io/tests/test_cellfinder_writer.py:
```diff
@@ -0,0 +1,24 @@
+    layers = reader_xml.xml_reader(xml_file)
+    assert len(layers) == 2
+
+    paths = writer_xml.write_multiple_points_to_xml(
+        str(tmpdir / "multiple.xml"), layers
+    )
```
I guess it should be able to. The file will [eventually just be opened with `open`](https://github.com/adamltyson/imlib/blob/625193be4a586d9040a48df9d51dbdd3a17c7d06/imlib/IO/cells.py#L130).
